workers/gossip-server/src/heartbeat.rs

```diff
@@ -63,105 +64,15 @@ impl GossipProtocolExecutor {
-        if peers_to_add.is_empty() {
-            return Ok(());
-        }
-
-        self.add_new_peers(&peers_to_add, incoming_peer_info).await?;
-        Ok(())
-    }
-
-    /// Merges the wallet information from an incoming heartbeat with the
-    /// locally stored wallet information
-    ///
-    /// In specific, the local peer must update its replicas list for any wallet
-    /// it manages
-    async fn merge_wallets(&self, peer_wallets: HashMap<WalletIdentifier, WalletMetadata>) {
-        let locked_wallets = self.global_state.read_wallet_index().await;
-        let locked_peers = self.global_state.read_peer_index().await;
-        for (wallet_id, mut wallet_info) in peer_wallets.into_iter() {
-            // Filter out any replicas that we don't have peer info for
-            // This may happen for a multitude of reasons; one reason is that the local node
-            // has expired a peer, but the remote node has not
-            //
-            // In this case, we leave the expired peer in the invisibility window, waiting
-            // for the expired peer to expire on all other cluster peers
-            wallet_info.replicas.retain(|replica| locked_peers.contains_peer(replica));
-
-            // Merge with the local copy of the wallet
-            locked_wallets.merge_metadata(&wallet_id, &wallet_info).await
-        }
-    }
-
-    /// Merges order book information from the incoming heartbeat request,
-    /// requests order information from peers if an order is not present
-    async fn merge_order_book(
-        &self,
-        incoming_orders: Vec<(OrderIdentifier, ClusterId)>,
-    ) -> Result<(), GossipError> {
-        // Build a list of orders not stored locally and request order information for
-        // each one
-        let mut new_orders = Vec::new();
-        {
-            let locked_order_book = self.global_state.read_order_book().await;
-            for (order_id, cluster) in incoming_orders.into_iter() {
-                if !locked_order_book.contains_order(&order_id) {
-                    new_orders.push((order_id, cluster));
-                }
-            }
-        } // locked_order_book released
-
-        // Request order information for all new orders
-        for (order_id, cluster) in new_orders.into_iter() {
-            // Pick a cluster peer to dial for the order info
-            if let Some(peer_id) =
-                self.global_state.read_peer_index().await.sample_cluster_peer(&cluster).await
-            {
-                self.network_channel
-                    .send(GossipOutbound::Request {
-                        peer_id,
-                        message: GossipRequest::OrderInfo(OrderInfoRequest { order_id }),
-                    })
-                    .map_err(|err| GossipError::SendMessage(err.to_string()))?;
+        let state = &self.global_state;
+        for (peer_id, info) in message.known_peers.into_iter() {
+            if !state.get_peer_info(&peer_id)?.is_none() {
```

Comment:
```suggestion
            if state.get_peer_info(&peer_id)?.is_none() {
```
right? if searching for the peer in the message returns a `None`, then add it

Reply:
good call!